drivers/overlay/ov_network.go

```diff
@@ -402,22 +402,24 @@ func (n *network) getBridgeNamePrefix(s *subnet) string {
 	return "ov-" + fmt.Sprintf("%06x", n.vxlanID(s))
 }
 
-func isOverlap(nw *net.IPNet) bool {
+func checkOverlap(nw *net.IPNet) error {
 	var nameservers []string
 
 	if rc, err := resolvconf.Get(); err == nil {
 		nameservers = resolvconf.GetNameserversAsCIDR(rc.Content)
 	}
 
 	if err := netutils.CheckNameserverOverlaps(nameservers, nw); err != nil {
-		return true
+		logrus.Errorf("overlay subnet %s has conflicts with nameservers %v", nw.String(), nameservers)
```

Comment:
Please remove the logs as requested by @mrjana. You can return a more expressive error if the one returned by the overlap check functions is not good enough.

Something like:

```
if err := netutils.CheckNameserverOverlaps(nameservers, nw); err != nil {
    return fmt.Errorf("overlay subnet %v failed check with name servers: %v: %v", nw, nameservers, err)
}
```